src/main/java/tutorease/address/logic/commands/DeleteLessonCommand.java

```diff
@@ -45,12 +46,12 @@ public CommandResult execute(Model model) throws CommandException {
 
         int listIndex = targetIndex.getZeroBased();
 
-        if (listIndex >= model.getLessonScheduleSize()) {
+        if (listIndex >= model.getFilteredLessonList().size()) {
```

Comment:
Yep we should use the filtered lesson list instead of the entire lesson schedule